### PR TITLE
Add onclick actions to time and sound blocks

### PR DESCRIPTION
**Pull request by OneMoreByte**
Adds the features @that1communist wanted in the clock and time blocks. So both #106 and #107 

Let me know if there is anything I can improve! I'm fairly new to rust

### CODE REVIEW

**Comment by atheriel** [comment]
It looks like there's some noise in these commits from `rustfmt`; the changes themselves are more confined. Here are my unsolicited review comments:

1. I would strongly prefer that the command run by the Sound block be configurable in the same way that it is for the Time block, instead of being hard-coded to run `pavucontrol`.
2. You should use `Option<String>` instead of `String` for the command configurations and handle `None`, instead of defaulting to `sleep 0`, which just seems wasteful.
3. I suggest the name `on_click` for the configuration options instead of the more ambiguous `command`.
4. Document any new configuration options for these blocks in `blocks.md`.

**Comment by OneMoreByte** [comment]
Alright, that shouldn't take to long. I'll get those fixed

**Comment by atheriel** [comment]
Your last commit deletes the `blocks.md` file, which I think is unintentional.

File: src/blocks/sound.rs [hunk 1]
```diff
@@ -228,23 +240,46 @@ impl Block for Sound {
     }
 
     fn click(&mut self, e: &I3BarEvent) -> Result<()> {
+
+
         if let Some(ref name) = e.name {
+            let mut command = "".to_string();
```
Comment (atheriel) [inline src/blocks/sound.rs]: This logic could probably all be moved into the match arm `MouseButton::Left => { ... }` below.

File: src/blocks/sound.rs [hunk 2]
```diff
@@ -115,6 +118,9 @@ pub struct SoundConfig {
     /// The steps volume is in/decreased for the selected audio device (When greater than 50 it gets limited to 50)
     #[serde(default = "SoundConfig::default_step_width")]
     pub step_width: u32,
+
+    #[serde(default = "SoundConfig::default_on_clicked")]
+    pub on_clicked: Option<String>,
```
Comment (atheriel) [inline src/blocks/sound.rs]: Can you change this to `on_click`, for consistency with the [Custom block](https://github.com/greshake/i3status-rust/blob/master/blocks.md#custom)?

File: src/blocks/time.rs [hunk 3]
```diff
@@ -30,6 +34,9 @@ pub struct TimeConfig {
     /// Update interval in seconds
     #[serde(default = "TimeConfig::default_interval", deserialize_with = "deserialize_duration")]
     pub interval: Duration,
+
+    #[serde(default = "TimeConfig::default_on_clicked")]
+    pub on_clicked: Option<String>,
```
Comment (atheriel) [inline src/blocks/time.rs]: Can you change this to `on_click`, for consistency with the [Custom block](https://github.com/greshake/i3status-rust/blob/master/blocks.md#custom)?

**Comment by OneMoreByte** [comment]
Oh geez, I didn't realize you'd get spammed with my git stuff. I'm kinda struggling with git right now. I'll fix that after I get everything on track again

**Comment by atheriel** [comment]
No worries, git lives up to its namesake. If you need suggestions I can try and be helpful.

**Comment by OneMoreByte** [comment]
Alright, I think I got it. Let me know If I broke anything. I'm going to get those things corrected quick

**Comment by atheriel** [comment]
I've merged this manually into master, but unfortunately GitHub doesn't realize it. Thanks!

**Pull request closed without merging**